src/core.js

```diff
@@ -1655,14 +1663,17 @@ define([
             var currentType = mug.__className,
                 questions = (mug.options.limitTypeChangeTo || 
                      _this.data.core.QUESTIONS_IN_TOOLBAR),
-                ret = [];
+                ret = [],
+                hasChildren = mug.form.getChildren(mug).length > 0,
+                isSelect = /^M?Select$/.test.bind(/^M?Select$/);
 
             for (var i = 0; i < questions.length; i++) {
                 var typeName = questions[i],
                     q = _this.data.core.mugTypes[typeName];
                 if (q.isTypeChangeable && currentType !== typeName &&
                     (!q.limitTypeChangeTo || 
-                     q.limitTypeChangeTo.indexOf(currentType) !== -1))
+                     q.limitTypeChangeTo.indexOf(currentType) !== -1) &&
+                    (!hasChildren || !isSelect(mug.__className) || isSelect(typeName)))
```

Comment:
I know I was the one to refactor similar code over [here](https://github.com/dimagi/Vellum/blob/limit-type-changes/src/mugs.js#L1080), but since then I have started to delegate this kind of decision to the mug so we don't have to hard-code type names in places like this. This also makes it easier for plugins like itemsets to write custom mug types that need to have a say in this decision. What do you think of adding a new mug option like this:

```js
if (q.isTypeChangeable && currentType !== typeName &&
    (!q.limitTypeChangeTo ||
     q.limitTypeChangeTo.indexOf(currentType) !== -1) &&
    mug.options.canChangeType(mug, typeName))
{
    ...
}
```
`defaultOptions` in mugs.js will have this:
```js
/**
 * Decide if this mug can have its type changed to typeName
 *
 * The value of `limitTypeChangeTo` takes precedence over this option.
 *
 * @param mug - the mug that would change type.
 * @param typeName - the name of the new type for mug (e.g., MSelect).
 * @returns - true if the mug can change to type, otherwise false.
 */
canChangeType: function (mug, typeName) { return true; },
```

Then MSelect, and by inheritance Select also, can provide a custom value for this option that checks if the mug has children, etc.